modules/33acrossIdSystem.js

```diff
@@ -95,14 +107,19 @@ export const thirthyThreeAcrossIdSubmodule = {
       callback(cb) {
         ajaxBuilder(AJAX_TIMEOUT)(apiUrl, {
           success(response) {
-            let envelope;
+            let responseObj = { };
 
             try {
-              envelope = getEnvelope(JSON.parse(response))
+              responseObj = calculateResponseObj(JSON.parse(response));
```

Comment:
Does it make more sense, logically, to put lines 117-122 in the try block? Also that way you do not have to separately declare and assign a value for `responseObj`. We know that `calculateResponseObj` will at the very least return a `{}`

Reply:
something like this?? -->
```
try {
  const responseObj = calculateResponseObj(JSON.parse(response));
  if (responseObj.ext) {
    storage.setDataInLocalStorage(EXT_KEY, JSON.stringify(responseObj.ext));
  }
  cb(responseObj.envelope);
} catch (err) {
  logError(`${MODULE_NAME}: ID reading error:`, err);
  cb();
}
```

it could be an option, but I feel that now the callback (cb) will be wrapped inside the try/catch and any exception triggered by the execution of that callback will be catched... and `ID reading error:` might not be the best error message for that exception.

Reply:
oh okay, currently errors related to cb are not handled at all and my suggestion would change that behavior. So nvm.